test/move_effect_attack_up_user_ally.c

```diff
@@ -0,0 +1,61 @@
+#include "global.h"
+#include "test_battle.h"
+
+ASSUMPTIONS
+{
+    ASSUME(gBattleMoves[MOVE_HOWL].effect == EFFECT_ATTACK_UP_USER_ALLY);
+}
+
+SINGLE_BATTLE_TEST("Howl raises user's Attack", s16 damage)
+{
+    bool32 raiseAttack;
+    PARAMETRIZE { raiseAttack = FALSE; }
+    PARAMETRIZE { raiseAttack = TRUE; }
+    GIVEN {
+        ASSUME(gBattleMoves[MOVE_TACKLE].split == SPLIT_PHYSICAL);
+        PLAYER(SPECIES_WOBBUFFET);
+        OPPONENT(SPECIES_WOBBUFFET);
+    } WHEN {
+        if (raiseAttack) TURN { MOVE(player, MOVE_HOWL); }
+        TURN { MOVE(player, MOVE_TACKLE); }
+    } SCENE {
+        if (raiseAttack) {
+            ANIMATION(ANIM_TYPE_MOVE, MOVE_HOWL, player);
+            ANIMATION(ANIM_TYPE_GENERAL, B_ANIM_STATS_CHANGE, player);
+            MESSAGE("Wobbuffet's attack rose!");
+        }
+        ANIMATION(ANIM_TYPE_MOVE, MOVE_TACKLE, player);
+        HP_BAR(opponent, captureDamage: &results[i].damage);
+    } FINALLY {
+        EXPECT_MUL_EQ(results[0].damage, Q_4_12(1.5), results[1].damage);
+    }
+}
+
+DOUBLE_BATTLE_TEST("Howl raises user's and partner's Attack", s16 damage)
+{
+    bool32 raiseAttack;
+    PARAMETRIZE { raiseAttack = FALSE; }
+    PARAMETRIZE { raiseAttack = TRUE; }
+    GIVEN {
+        ASSUME(gBattleMoves[MOVE_TACKLE].split == SPLIT_PHYSICAL);
+        PLAYER(SPECIES_WOBBUFFET) { Speed(15); };
+        PLAYER(SPECIES_WYNAUT) { Speed(10); };
+        OPPONENT(SPECIES_WOBBUFFET) { Speed(13); };
+        OPPONENT(SPECIES_WYNAUT) { Speed(12); };
+    } WHEN {
+        if (raiseAttack) TURN { MOVE(playerLeft, MOVE_HOWL); }
+        TURN { MOVE(playerLeft, MOVE_TACKLE, target: opponentLeft); }
```

Comment:
I think it might be good to also have the `playerRight` attack `opponentRight` and capture that damage (in a second `HP_BAR`), and check the damage was increased (in a second `EXPECT_MUL_EQ`).